### ISSUE TITLE
v0.29.1 recieving a throw on get task when fetching a task of an index creation

### ISSUE THREAD

**Issue by bidoubiwa** [meilisearch/meilisearch#2731]
**Describe the bug**

When I dont have the rights to create a specific key, but I create it anyway, the returned taskUid is considered not existing.

The task of the index creation is the following 

Key: 

```js
      const key = await client.createKey({
        uid,
        description: 'Indexing Products API key',
        actions: ['indexes.*', 'tasks.*'],
        indexes: ['index_1'],
        expiresAt: null,
      })
```

```js
      const newClient = new MeiliSearch({ host: HOST, apiKey: key.key })
      const taskInfo = await newClient.createIndex('indexes_wildcards')
```

Returned task
```
   {
      taskInfo: {
        taskUid: 3758,
        indexUid: 'indexes_wildcards',
        status: 'enqueued',
        type: 'indexCreation',
        enqueuedAt: '2022-09-01T11:12:33.610232Z'
      }
    }
```

When making a query on 
```
GET /tasks/3758
```

I receive the following error: 

```
Task `3758` not found.
```

Meilisearch log

```
[2022-09-01T11:16:20Z INFO  actix_web::middleware::logger] 127.0.0.1 "GET /tasks/3758 HTTP/1.1" 404 142 "-" "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)" 0.000211
```

**To Reproduce**
- Create an index on which the key has no right
- Fetch task that was returned by the index creation route
- Error, the task does not exists

**Expected behavior**
No provided task uid should be not found.

**Meilisearch version:** 
v0.29.1

**Comment by ManyTheFish** [comment]
I managed to reproduce this bug, it will be fixed in the v0.29.0rc3

**Comment by curquiza** [comment]
Fix by #2734 will be released in v0.29.0rc3
